server/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -333,4 +342,36 @@ static void selfTest() throws IOException {
             throw new SecurityException("Security misconfiguration: cannot access java.io.tmpdir", problem);
         }
     }
+
+    /**
+     * Prepopulates the system's security manager setters map with this class as a caller.
+     * This is loathsome, but avoids the annoying warning message at run time.
+     */
+    static void prepopulateSecurityCaller() {
+        try {
+            Class<?> c = Class.forName("sun.misc.Unsafe");
+            MethodHandles.Lookup lookup = MethodHandles.privateLookupIn(c, MethodHandles.lookup());
```

Comment:
I just had one minor question for my own curiosity, does this work exactly the same for all JDK versions? Does it make sense for us to do this only for JDK17 where the warning is shown? I'm thinking more along the lines of 7.16 and the various supported JDK versions we run/test with.

Reply:
This is an excellent point. The intent is that the code is effectively optional, it should be benign where the JDK 17 specific field `caller` is not present. Given that 8.0 runs on >= 17, I didn't think too hard about this, but given the potential backport to 7.16.x I'll double check this (maybe add a test too).